src/osp/Active/SysAreaAssociate.h

```diff
@@ -34,125 +34,120 @@
 namespace osp::active
 {
 
-class SysAreaAssociate;
-struct ACompActivatedSat;
+struct ActivationTracker
+{
+    using MapSatToEnt_t = std::unordered_map<universe::Satellite, ActiveEnt>;
+
+    // Satellites that are currently inside the active area
+    // possibly replace with a single entt sparse_set in ACompAreaLink
+    MapSatToEnt_t m_inside;
 
-struct StatusActivated
+    std::vector<MapSatToEnt_t::iterator> m_enter;
+    std::vector<std::pair<universe::Satellite, ActiveEnt>> m_leave;
+};
+
+struct ACompAreaLink
 {
-    int m_status; // 0 for no errors
-    ActiveEnt m_entity;
-    bool m_mutable;
+    ACompAreaLink(universe::Universe& rUniverse, universe::Satellite areaSat)
+     : m_areaSat(areaSat)
+     , m_rUniverse(rUniverse)
+     , m_activated(rUniverse.sat_type_count())
+    { }
+
+    ActivationTracker& get_tracker(universe::TypeSatIndex type)
+    { return m_activated[std::size_t(type)]; }
+
+    universe::Universe& get_universe() noexcept
+    { return m_rUniverse.get(); }
+
+    universe::Satellite m_areaSat;
```

Comment:
I've never really looked at how the sys area associate stuff works. So is the area itself a satellite? It's associated more closely to the universe than to the scene?